src/lua/alloc.c

```diff
@@ -0,0 +1,150 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2024, Tarantool AUTHORS, please see AUTHORS file.
+ */
+#include "lua/utils.h"
+
+#include <lauxlib.h> /* luaL_error */
+
+#define LUA_MEMORY_DEFAULT (2ULL * 1024 * 1024 * 1024)
+
+/**
+ * The default allocator function which is wrapped into a new one
+ * with the Lua memory limit checker.
+ */
+static lua_Alloc orig_alloc;
+/**
+ * The maximum allowed LuaJIT memory limit.
+ */
+static size_t memory = LUA_MEMORY_DEFAULT;
+/**
+ * Currently allocated memory amount by LuaJIT.
+ */
+static size_t allocated;
+
+/**
+ * Lua custom memory allocation function. It extends the original
+ * one with a memory counter and a limit check.
+ */
+static void *
+luaT_alloc(void *ud, void *ptr, size_t osize, size_t nsize)
+{
+	size_t new_allocated = allocated + nsize - osize;
+	if (new_allocated > memory) {
+		/* Returning NULL results with "not enough memory" error. */
```

Comment:
Nit: Comment line width is more than 66 symbols.
Typo: s/results with/results in the/

Reply:
Thank you for noticing the typo.

But for the line width: it's less than 66. Or one tab counts as 8 spaces? If so, I've fixed it.
And, actually, I don't understand why the limit should be 66. I see no such point in [https://www.tarantool.io/en/doc/latest/contributing/c_style_guide/](https://www.tarantool.io/en/doc/latest/contributing/c_style_guide/).

Reply:
66 characters limit was enforced for comments in the past as optimal width for a prose text readability. But it was not about a significant columns, includes the indent. Don't know why. Later we agreed that 80 symbols is good too.

Personally, I continue to use 66, but it is not enforced anymore.

I guess Sergey refers to the old limit.

Reply:
OK, maybe it is worth updating the Wiki then (I'm personally afraid to alter Vlad's manual :) ).